scapy/sessions.py

```diff
@@ -342,13 +344,17 @@ def _process_packet(self, pkt):
         return None
 
     def on_packet_received(self, pkt):
-        # type: (Optional[Packet]) -> None
+        # type: (Optional[_UniPacketList]) -> None
         """Hook to the Sessions API: entry point of the dissection.
         This will defragment IP if necessary, then process to
         TCP reassembly.
         """
         if not pkt:
             return None
+        if isinstance(pkt, list) or isinstance(pkt, PacketList):
+            for p in pkt:
+                self.on_packet_received(p)
+            return
```

Comment:
I'm not a huge fan of this. I know it's somewhat implemented in `DefaultSession`, but the point of `Session` was to handle packet storage. I would have prefered if whoever was calling `on_packet_received` with a packet list called it several times instead of adding this logic.

Reply:
I didn't found this mechanism used anywhere. I removed it.